library/cohesity_vmware_job.py

```diff
@@ -0,0 +1,472 @@
+    :return:
+    '''
+    try:
+        cluster_vip = module.params.get('cluster')
+        username = module.params.get('username')
+        password = module.params.get('password')
+        domain = 'LOCAL'
+        if "@" in username:
+            user_domain = username.split("@")
+            username = user_domain[0]
+            domain = user_domain[1]
+        cohesity_client = CohesityClient(cluster_vip=cluster_vip,
+                                         username=username,
+                                         password=password,
+                                         domain=domain)
+        return cohesity_client
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def main():
+    argument_spec = dict(
+            cluster=dict(type='str', required=True),
+            username=dict(type='str', required=True),
+            password=dict(type='str', required=True, no_log=True),
+            job_name=dict(type='str', required=True),
+            description=dict(type='str'),
+            state=dict(type='str', choices=['present', 'absent', 'started', 'stopped'], default='present'),
+            ondemand_run_type=dict(type='str', choices=['Regular', 'Full', 'Log', 'System'], default='Regular'),
+            wait_minutes=dict(type='int', default=30),
+            protection_source=dict(type='str', required=True),
+            timezone=dict(type='str', default='America/Los_Angeles'),
+            include=dict(type='list', elements='str'),
+            exclude=dict(type='list', elements='str'),
+            protection_policy=dict(type='str', required=True),
+            storage_domain=dict(type='str', required=True),
+            start_time=dict(type='str'),
+            delete_snapshots=dict(type='bool', default=False)
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True)
+    results = dict(
+        changed=False,
+        msg="Managing Cohesity protection job",
+        state=module.params.get('state')
+    )
+
+    global cohesity_client
+    cohesity_client = get_cohesity_client(module)
+    job_details = get_vmware_job_details(module)
+
+    if module.check_mode:
+        if module.params.get('state') == "present":
+            if job_details:
+                check_mode_results = dict(
+                    changed=False,
+                    msg="Check Mode: The Cohesity protection job already exists on the cluster",
+                )
+            else:
+                check_mode_results = dict(
+                    changed=True,
+                    msg="Check Mode: The Cohesity protection job is created",
+                )
+        elif module.params.get('state') == "absent":
+            if job_details:
+                check_mode_results = dict(
+                    changed=True,
+                    msg="Check Mode: The Cohesity protection job is deleted",
+                )
```

Comment:
this should have ended after each dict end, its ok for now. next time take care of this.

Reply:
ok, will change it now